### PR TITLE
LPS-94713 | Apply new Portlet URL utilities in Collaboration modules

### PR DESCRIPTION
**Pull request by kresimir-coko**
Hey Ambrin, @jbalsas told me to send this one to you when I fix the missing `.toString()`. 

One thing that maybe isn't expected here is the commit related to the `frontend-js-web` module that just handles slight renaming.

### CODE REVIEW

File: modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/global.es.js [hunk 1]
```diff
@@ -80,7 +80,7 @@ Liferay.Util.PortletURL = {
 	createActionURL,
 	createRenderURL,
 	createResourceURL,
-	createURL
```
Comment (jbalsas) [inline modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/global.es.js]: This should fail sf now, you probably need to sort it

File: modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/global.es.js [hunk 2]
```diff
@@ -80,7 +80,7 @@ Liferay.Util.PortletURL = {
...
+	createPortletURL
```
Comment (jbalsas) [inline modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/global.es.js]: This should fail sf now, you probably need to sort it

File: modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/global.es.js [hunk 3]
```diff
@@ -21,7 +21,7 @@ import {
 	getComponentCache,
 	initComponentCache
 } from './component.es';
-import createURL from './util/portlet_url/create_portlet_url.es';
+import createPortletURL from './util/portlet_url/create_portlet_url.es';
```
Comment (jbalsas) [inline modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/global.es.js]: You might want to sort this too

File: modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/global.es.js (hunk 2 above)
Comment (kresimir-coko) [reply modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/global.es.js]: I have become too reliant on auto-formatting done by VSCode and Prettier

Comment (jbalsas) [reply modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/global.es.js]: `sort-object-keys` is not an autofixable rule. In any case, you running `yarn format` should be a good sanity check always.

**Comment by jbalsas** [comment]
ci:test:sf

**Comment by kresimir-coko** [comment]
I've force pushed SF changes @ambrinchaudhary @jbalsas

**Comment by ambrinchaudhary** [comment]
ci:test:sf

**Comment by ambrinchaudhary** [comment]
ci:test:relevant

File: modules/apps/sharing/sharing-web/src/main/resources/META-INF/resources/dynamic_include/bottom.jsp [hunk 4]
```diff
@@ -59,28 +59,38 @@ sharingURL.setWindowState(LiferayWindowState.POP_UP);
 		Sharing,
 		'share',
 		function(classNameId, classPK, title, namespace, refreshOnClose) {
-			var sharingURL = new Liferay.PortletURL.createURL('<%= sharingURL.toString() %>');
+			var parameters = {
+				'classNameId': classNameId,
+				'classPK': classPK
+			}
```
Comment (jbalsas) [inline modules/apps/sharing/sharing-web/src/main/resources/META-INF/resources/dynamic_include/bottom.jsp]: Missing semicolon here?

We might also want to get rid of the quotes for the object keys as they don't seem necessary...

File: modules/apps/sharing/sharing-web/src/main/resources/META-INF/resources/dynamic_include/bottom.jsp [hunk 5]
```diff
@@ -59,28 +59,38 @@ sharingURL.setWindowState(LiferayWindowState.POP_UP);
...
 
 	Liferay.provide(
 		Sharing,
 		'manageCollaborators',
 		function(classNameId, classPK, namespace, refreshOnClose) {
-			var manageCollaboratorsURL = new Liferay.PortletURL.createURL('<%= manageCollaboratorsURL.toString() %>');
+			var parameters = {
+				'classNameId': classNameId,
+				'classPK': classPK
+			}
```
Comment (jbalsas) [inline modules/apps/sharing/sharing-web/src/main/resources/META-INF/resources/dynamic_include/bottom.jsp]: Semicolon?

File: modules/apps/opensocial/opensocial-portlet/docroot/gadget/js/main.js [hunk 6]
```diff
@@ -23,576 +23,624 @@ AUI.add(
...
-					checksum: {},
-					content: {},
-					country: {
-						setter: function(v) {
-							if (!v) {
-								v = 'ALL';
-							}
-
-							return v;
+		var TPL_IFRAME =
```
Comment (jbalsas) [inline modules/apps/opensocial/opensocial-portlet/docroot/gadget/js/main.js]: This module is:

```
Liferay-Releng-Bundle: false
Liferay-Releng-Deprecated: true
Liferay-Releng-Marketplace: false
```

I wouldn't even bother. I'd say we can just drop this commit and let it be.

Comment (kresimir-coko) [reply modules/apps/opensocial/opensocial-portlet/docroot/gadget/js/main.js]: I wonder how many of these deprecated modules I've worked on in the past couple months without even knowing they were deprecated.

Gonna have to make it a habit of checking the `app.bnd` file before working on any of the less obvious ones.

Comment (kresimir-coko) [reply modules/apps/opensocial/opensocial-portlet/docroot/gadget/js/main.js]: Do we just forget this entire PR ever happened?

Comment (jbalsas) [reply modules/apps/opensocial/opensocial-portlet/docroot/gadget/js/main.js]: No, just drop the commits for the `opensocial` and `microblogs` app which are the ones that don't really need it. The `sharing` app is still in development, so it's worth pushing it through.

Comment (jbalsas) [reply modules/apps/opensocial/opensocial-portlet/docroot/gadget/js/main.js]: Or resend, as you see fit :)

Comment (kresimir-coko) [reply modules/apps/opensocial/opensocial-portlet/docroot/gadget/js/main.js]: I'll remove the code related to `opensocial` and `microblogs` and force push the remaining code to this PR

**Comment by kresimir-coko** [comment]
@ambrinchaudhary Here, I hope we don't need to change anything else. May the force push be with you

**Comment by ambrinchaudhary** [comment]
Just started reviewing :)

**Comment by ambrinchaudhary** [comment]
Pull request submitted to https://github.com/brianchandotcom/liferay-portal/pull/77324.

**Pull request closed without merging**